File: netcore/Flights/Flights.Core/Interfaces/IFlightService.cs
using Flights.Core.Model;
using System.Collections.Generic;

namespace Flights.Core.Interfaces
{
    public interface IFlightService
    {
        Flight GetFlight(string id);
        List<Flight> GetFlights(string filter);
        void InsertFlight(Flight flight);
        void UpdateFlight(Flight flight);
        void DeleteFlight(string id);
    }
}

File: Java/src/main/java/flights/core/interfaces/IFlightService.java
package flights.core.interfaces;

import flights.core.model.Flight;
import java.util.List;

public interface IFlightService {
  Flight getFlight(String id);
  List<Flight> getFlights(String filter);
  Flight insertFlight(Flight flight);
  Flight updateFlight(Flight flight);
  void deleteFlight(String id);
}
